src/main/java/slimeknights/tconstruct/library/recipe/material/MaterialCastingRecipe.java

```diff
@@ -0,0 +1,71 @@
+package slimeknights.tconstruct.library.recipe.material;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import slimeknights.tconstruct.library.MaterialRegistry;
+import slimeknights.tconstruct.library.materials.IMaterial;
+import slimeknights.tconstruct.library.materials.MaterialId;
+import slimeknights.tconstruct.library.recipe.RecipeTypes;
+import slimeknights.tconstruct.smeltery.TinkerSmeltery;
+import slimeknights.tconstruct.smeltery.recipe.ICastingInventory;
+
+@AllArgsConstructor
+public class MaterialCastingRecipe implements IRecipe<ICastingInventory> {
```

Comment:
Not a fan of a whole additional recipe type here. The material type was added because each material has multiple possible inputs of varying amounts and we need it in both part casting and tool repair. Fluids we only ever have one for each material (if you ignore tags).  Just use `IMaterial::getFluid()`